### PR TITLE
build: Require pkg-config for all of the hosts

### PR DESCRIPTION
**Pull request by hebasto**
This PR:
- is based on #18297 (already merged)
- drops all of the non-pkg-config paths from the `configure` script

Ref: #17768

### CODE REVIEW

**Comment by laanwj** [comment]
I like this idea, especially the simplification of the autoconf scripts. Concept ACK.

**Comment by hebasto** [comment]
Updated 284418ad2446fec21546e485ec0af0edde2eb773 -> b3b769adf1868b2c3ebdd1d9778fac58125e2b8d ([pr18307.01](https://github.com/hebasto/bitcoin/commits/pr18307.01) -> [pr18307.02](https://github.com/hebasto/bitcoin/commits/pr18307.02), [diff](https://github.com/hebasto/bitcoin/compare/pr18307.01..pr18307.02)):

- rebased due to the conflict with #18298

**Comment by theuni** [comment]
Concept ACK, thanks for working on this. Will review the chain.

**Comment by hebasto** [comment]
Rebased b3b769adf1868b2c3ebdd1d9778fac58125e2b8d -> a4535bfd4d47538db47b9787da8922cd06bd6fd7 ([pr18307.02](https://github.com/hebasto/bitcoin/commits/pr18307.02) -> [pr18307.03](https://github.com/hebasto/bitcoin/commits/pr18307.03)) due to the merging of #18297.

**Comment by hebasto** [comment]
@fanquake 
May I ask "Needs gitian build" and "Needs Guix build" labels for this PR?

**Review by fanquake — COMMENTED** [review]
Concept ACK - This looks pretty good, and is a nice simplification. This should be a no-op for most platforms, besides some reordering of package checks in configure. Obviously Windows is different, and after this PR, should be full pkg-config. This is the main difference I'm seeing in `config.log`:
```diff
# libevent
-EVENT_CFLAGS=''
-EVENT_LIBS='-levent'
+EVENT_CFLAGS='-I/bitcoin/depends/x86_64-w64-mingw32/include'
+EVENT_LIBS='-L/bitcoin/depends/x86_64-w64-mingw32/lib -levent -lws2_32'
# qrencode
-QR_CFLAGS=''
-QR_LIBS='-lqrencode'
+QR_CFLAGS='-I/bitcoin/depends/x86_64-w64-mingw32/include'
+QR_LIBS='-L/bitcoin/depends/x86_64-w64-mingw32/lib -lqrencode -lpthread'
# zeromq
-ZMQ_CFLAGS=' -DZMQ_STATIC'
-ZMQ_LIBS='-lzmq'
+ZMQ_CFLAGS='-I/bitcoin/depends/x86_64-w64-mingw32/include -DZMQ_STATIC'
+ZMQ_LIBS='-L/bitcoin/depends/x86_64-w64-mingw32/lib -lzmq -liphlpapi -lpthread'
```

`liphlpapi` was added to libzmqs `PKGCFG_LIBS_PRIVATE`, when static linking, in https://github.com/zeromq/libzmq/pull/2787.

File: configure.ac [hunk 1]
```diff
@@ -1301,115 +1292,66 @@ CPPFLAGS="$TEMP_CPPFLAGS"
...
-  if test x$build_bitcoin_cli$build_bitcoind$bitcoin_enable_qt$use_tests$use_bench != xnonononono; then
-    AC_CHECK_HEADER([event2/event.h], [use_libevent=yes], AC_MSG_ERROR(libevent headers missing),)
-    AC_CHECK_LIB([event],[main],EVENT_LIBS=-levent,AC_MSG_ERROR(libevent missing))
-    if test x$TARGET_OS != xwindows; then
-      AC_CHECK_LIB([event_pthreads],[main],EVENT_PTHREADS_LIBS=-levent_pthreads,AC_MSG_ERROR(libevent_pthreads missing))
-    fi
+if test x$build_bitcoin_cli$build_bitcoind$bitcoin_enable_qt$use_tests$use_bench != xnonononono; then
+  PKG_CHECK_MODULES([EVENT], [libevent >= 2.0.21], [use_libevent=yes], [AC_MSG_ERROR(libevent version 2.0.21 or greater not found.)])
+  if test x$TARGET_OS != xwindows; then
+    PKG_CHECK_MODULES([EVENT_PTHREADS], [libevent_pthreads],, AC_MSG_ERROR([libevent_pthreads not found]))
```
[the remark this answers is not in the document]
Comment (hebasto) [reply configure.ac:1300]: > Was removing the version checking for `libevent_pthreads` intentional?

No, it seems like a rebasing glitch. Going to fix.
Thanks!

**Comment by hebasto** [comment]
Updated a4535bfd4d47538db47b9787da8922cd06bd6fd7 -> 4e006b640eb94d0699169b114842aca3b6ce585a ([pr18307.03](https://github.com/hebasto/bitcoin/commits/pr18307.03) -> [pr18307.05](https://github.com/hebasto/bitcoin/commits/pr18307.05), [diff](https://github.com/hebasto/bitcoin/compare/pr18307.03..pr18307.05)):

- addressed @fanquake's [comment](https://github.com/bitcoin/bitcoin/pull/18307#discussion_r443360536):
> Was removing the version checking for `libevent_pthreads` intentional?

File: configure.ac (hunk 1 above)
[the remark this answers is not in the document]
Comment (hebasto) [reply configure.ac]: [Updated](https://github.com/bitcoin/bitcoin/pull/18307#issuecomment-647371783).

**Comment by hebasto** [comment]
Updated 4e006b640eb94d0699169b114842aca3b6ce585a -> 92bc268e4af4ebcbde08567ea00e019ac509a769 ([pr18307.05](https://github.com/hebasto/bitcoin/commits/pr18307.05) -> [pr18307.06](https://github.com/hebasto/bitcoin/commits/pr18307.06), [diff](https://github.com/hebasto/bitcoin/compare/pr18307.05..pr18307.06)):

- fixed conflict with #18677

**Comment by laanwj** [comment]
ACK 92bc268e4af4ebcbde08567ea00e019ac509a769
Changes look correct, and I checked that there are no remaining matches for `use_pkgconfig` after this.

File: configure.ac [hunk 2]
```diff
@@ -1301,115 +1292,66 @@ CPPFLAGS="$TEMP_CPPFLAGS"
...
+    AC_MSG_WARN([libzmq version 4.x or greater not found, disabling])
+    use_zmq=no])
+else
+  AC_DEFINE_UNQUOTED([ENABLE_ZMQ], [0], [Define to 1 to enable ZMQ functions])
+fi
+
+if test "x$use_zmq" = xyes; then
+  dnl Assume libzmq was built for static linking
+  case $host in
+    *mingw*)
```
Comment (laanwj) [inline configure.ac:1325]: It's somewhat unfortunate that this mingw-specific workaround is still ncessary, that the required flag is not passed in through the pkgconfig file in the case of a static build. But I suppose this is an upstream issue.

Comment (hebasto) [reply configure.ac:1325]: Related to #8238.

**Review by fanquake — APPROVED** [review]
ACK 92bc268e4af4ebcbde08567ea00e019ac509a769. I re-gitian-built. There are a couple follow-ups that I'll PR shortly. Thanks for addressing my feedback above. I took too long to get back to this.

**Pull request merged**